core/src/apps/common/paths.py

```diff
@@ -372,3 +372,42 @@ def _path_item(i: int) -> str:
 
 def unharden(item: int) -> int:
     return item ^ (item & HARDENED)
+
+
+def get_account_num(address_n: list[int], pattern: str | list[str]) -> int:
+    if isinstance(pattern, str):
+        pattern = [pattern]
+
+    # Trying all possible patterns - at least one should match
+    for patt in pattern:
+        try:
+            return _get_account_num_single(address_n, patt)
+        except ValueError:
+            pass
+
+    # This function should not raise
+    return 0
```

Comment:
We should return None instead and not display account number at all in this case.

Reply:
Done in [b558b39](https://github.com/trezor/trezor-firmware/pull/3299/commits/b558b39a7128223d023b95140d0257bf93e1261f) - a whole new helper function doing this logic